variable/include/scipp/variable/buckets.h

```diff
@@ -19,4 +19,8 @@ SCIPP_VARIABLE_EXPORT void copy_slices(const VariableConstView &src,
 [[nodiscard]] SCIPP_VARIABLE_EXPORT std::tuple<Variable, scipp::index>
 sizes_to_begin(const VariableConstView &sizes);
 
+[[nodiscard]] SCIPP_VARIABLE_EXPORT Variable from_constituents(Variable indices,
```

Comment:
So passing by value is intentional here? I think i remember seeing this function's body using `move` somewhere above, but I am not sure anymore.

Reply:
Yes, passing by value gives us both, copy and move support:

```cpp
Variable var;
from_constituents(var); // makes a copy
from_constituents(std::move(var)); // moves, no copy
```

Passing by value is a way to avoid overloading on `&&`. The downside is that we have to move *twice*, but in practice this does not matter for large objects (it would if designing a class that accepts, e.g., small strings).